### PR TITLE
[Merged by Bors] - chore: Reduce imports to `Data.Fin.Basic`

### PR DESCRIPTION
**Pull request by YaelDillies**
By using `Nat` and `Int` specific lemmas, we can reduce the import to `Data.Fin.Basic` to almost nothing.

I haven't gone all the way to removing all dependencies because that involves moving declarations around and I would like to keep the diff clean.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fin/Basic.lean [hunk 1]
```diff
@@ -1647,7 +1647,7 @@ theorem add_one_le_of_lt {n : ℕ} {a b : Fin (n + 1)} (h : a < b) : a + 1 ≤ b
 
 theorem exists_eq_add_of_le {n : ℕ} {a b : Fin n} (h : a ≤ b) : ∃ k ≤ b, b = a + k := by
   obtain ⟨k, hk⟩ : ∃ k : ℕ, (b : ℕ) = a + k := Nat.exists_eq_add_of_le h
-  have hkb : k ≤ b := le_add_self.trans hk.ge
+  have hkb : k ≤ b := by omega
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Fin/Basic.lean:1650]:
```suggestion
  have hkb : k ≤ b := (Nat.le_add_left _ _).trans hk.ge
```

File: Mathlib/Data/Fin/Basic.lean [hunk 2]
```diff
@@ -1111,8 +1113,7 @@ theorem coe_succ_lt_iff_lt {n : ℕ} {j k : Fin n} : (j : Fin <| n + 1) < k ↔
 
 @[simp]
 theorem range_castSucc {n : ℕ} : Set.range (castSucc : Fin n → Fin n.succ) =
-    ({ i | (i : ℕ) < n } : Set (Fin n.succ)) :=
-  range_castLE le_self_add
+    ({ i | (i : ℕ) < n } : Set (Fin n.succ)) := range_castLE (by omega)
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Fin/Basic.lean:1116]:
```suggestion
    ({ i | (i : ℕ) < n } : Set (Fin n.succ)) :=
  range_castLE (Nat.le_add_right _ _)
```

File: Mathlib/Data/Fin/Basic.lean [hunk 3]
```diff
@@ -1656,12 +1656,10 @@ theorem exists_eq_add_of_lt {n : ℕ} {a b : Fin (n + 1)} (h : a < b) :
...
   · cases' a with a ha
     cases' b with b hb
-    simp only [Nat.zero_eq, zero_add, Nat.lt_one_iff] at ha hb
+    simp only [Nat.zero_eq, Nat.zero_add, Nat.lt_one_iff] at ha hb
     simp [ha, hb] at h
   obtain ⟨k, hk⟩ : ∃ k : ℕ, (b : ℕ) = a + k + 1 := Nat.exists_eq_add_of_lt h
-  have hkb : k < b := by
-    rw [hk, add_comm _ k, Nat.lt_succ_iff]
-    exact le_self_add
+  have hkb : k < b := by omega
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Fin/Basic.lean:1662]: Same

**Review by Ruben-VandeVelde — COMMENTED** [review]
I like it, though I'd suggest using the equivalent Nat lemmas rather than omega.

File: Mathlib/Data/Fin/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Fin/Basic.lean:1650]: Why? `omega` is instantaneous.

**Comment by YaelDillies** [comment]
Maybe @semorrison can weigh in 😉

**Comment by semorrison** [comment]
`omega` isn't going anywhere. I think it's fine to use here.

**Comment by semorrison** [comment]
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**